nncf/quantization/layers.py

```diff
@@ -232,16 +235,57 @@ def broadcast_initialized_params(self, src: int = 0):
     def set_export_mode(self, mode: QuantizerExportMode):
         self._export_mode = mode
 
-    def run_export_quantization(self, x: torch.Tensor):
-        raise NotImplementedError
-
     def extra_repr(self):
         return 'bit={}, ch={}, wt={}'.format(
             self.num_bits, self.per_channel, self.is_weights)
 
+    def _get_input_low_input_high(self, **kwargs):
+        raise NotImplementedError
+
     def get_current_config(self) -> QuantizerConfig:
         raise NotImplementedError
 
+    def _prepare_export_quantization(self, x: torch.Tensor):
+        raise NotImplementedError
+
+    def _prepare_fq_export_quantization(self, x: torch.Tensor):
+        x, level_high, level_low, input_low, input_high = self._prepare_export_quantization(x)
+        with no_jit_trace():
+            levels = level_high - level_low
+        return x, levels, input_low, input_high
+
+    def _prepare_qdq_export_quantization(self, x: torch.Tensor):
+        x, level_high, level_low, input_low, input_high = self._prepare_export_quantization(x)
+        with no_jit_trace():
+            y_scale, y_zero_point = get_scale_zp_from_input_low_input_high(level_low,
+                                                                           level_high,
+                                                                           input_low,
+                                                                           input_high)
+        return x, y_scale, y_zero_point
+
+    def run_export_quantization(self, x: torch.Tensor):
+        if self._export_mode == QuantizerExportMode.FAKE_QUANTIZE:
+            x, levels, input_low, input_high = self._prepare_fq_export_quantization(x)
+            return ExportQuantizeToFakeQuantize.apply(x, levels,
+                                                      input_low,
+                                                      input_high,
+                                                      input_low,
+                                                      input_high)
+        elif self._export_mode == QuantizerExportMode.ONNX_QUANTIZE_DEQUANTIZE_PAIRS:
+            x, y_scale, y_zero_point = self._prepare_qdq_export_quantization(x)
+            if self.per_channel:
+                try:
+                    if torch.allclose(y_scale - y_scale[0], torch.zeros_like(y_scale)) and \
+                            torch.allclose(y_zero_point - y_zero_point[0], torch.zeros_like(y_zero_point)):
+                        y_scale, y_zero_point = y_scale[0], y_zero_point[0]
+                        return ExportQuantizeToONNXQuantDequant.apply(x, y_scale, y_zero_point)
+                    raise RuntimeError("PyTorch v1.5.0 export to ONNX using QuantizeLinear-DequantizeLinear "
+                                       "doesn't support per channel quantization")
+                except IndexError:
+                    return ExportQuantizeToONNXQuantDequant.apply(x, y_scale, y_zero_point)
```

Comment:
Exception handling is not necessary here. Rewrite using `if`'s and either `y_scale.shape()`, or `self.scale_shape`

Reply:
Updated; please, check it out